src/Front/Parser.y

```diff
@@ -270,16 +281,12 @@ exprPar                               --> [Expr]
   | expr                                { [$1] }
 {
 -- | What to do upon encountering parse error.
---
--- TODO does this really just throw an uncaught exception on parse error, or
--- does Happy catch that for us?
-parseError :: Token -> a
-parseError (Token (sp, _)) =
-    error $ show (pretty sp) ++ ":Syntax error"
+parseError :: Token -> Alex a
+parseError (Token (sp, _)) = syntaxErr $ "near " ++ show (pretty sp)
 
 -- | Parse a 'String' and yield a 'Program' or an 'ErrorMsg' if unsuccessful.
```

Comment:
is this docstring still accurate? that was obvious to me when it was `Either ErrorMsg Program`.